### PR TITLE
Add pluggable auth support for external account creds

### PR DESCRIPTION
**Pull request by mario-vimal**

### CODE REVIEW

File: src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
+
+//
+// Copyright 2020 gRPC authors.
```
Comment (aeitzman) [inline src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc]: super nit: 2023

File: src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#include <grpc/support/string_util.h>
+
+#include "src/core/lib/gprpp/env.h"
+#include "src/core/lib/http/httpcli_ssl_credentials.h"
+#include "src/core/lib/iomgr/closure.h"
+#include "src/core/lib/json/json.h"
+#include "src/core/lib/json/json_reader.h"
+#include "src/core/lib/json/json_writer.h"
+#include "src/core/lib/security/credentials/credentials.h"
+#include "src/core/lib/uri/uri_parser.h"
```
Comment (aeitzman) [inline src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc]: nit: Do we need all these includes? Wondering about the uri_parser in particular

File: src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    *error = GRPC_ERROR_CREATE("command field must be a string.");
+    return;
+  }
+  command_ = executable_it->second.string();
+  executable_timeout_ms_ = DEFAULT_EXECUTABLE_TIMEOUT_MS;
+  executable_it = executable_json.object().find("timeout_millis");
+  if (executable_it != executable_json.object().end()) {
+    if (!absl::SimpleAtoi(executable_it->second.string(),
+                          &executable_timeout_ms_)) {
+      *error = GRPC_ERROR_CREATE("timeout_millis field must be a number");
```
Comment (aeitzman) [inline src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc]: nit: add period at the end of the error message

File: src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc (hunk 2 above)
Comment (mario-vimal) [reply src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc]: We need not worry about includes as the "include what you use" iwyu tool will show us the unused and missed includes during build.
I've modified the includes for now by running iwyu locally, which isn't that reliable.

Comment (aeitzman) [reply src/core/lib/security/credentials/external/pluggable_auth_external_account_credentials.cc]: oh great, no worries then

File: test/core/security/credentials_test.cc [hunk 4]
```diff
@@ -3799,6 +3800,113 @@ TEST(CredentialsTest,
...
+  auto creds =
+      ExternalAccountCredentials::Create(*options_json, {"scope"}, &error);
+  std::string actual_error, expected_error = "command field not present.";
+  grpc_error_get_str(error, StatusStrProperty::kDescription, &actual_error);
+  GPR_ASSERT(strcmp(actual_error.c_str(), expected_error.c_str()) == 0);
+}
+
+TEST(
+    CredentialsTest,
+    TestPluggableAuthExternalAccountCredentialsCreateFailureExecutableTimeoutLessThanMin) {
```
Comment (aeitzman) [inline test/core/security/credentials_test.cc]: nit: less than 5 seconds?

**Review by aeitzman — APPROVED** [review]

**Pull request merged**